### ISSUE TITLE
make `ppx_tools` an optional dep in OPAM

### ISSUE THREAD

**Issue by lefessan** [ocsigen/lwt#266]
I recently tried to install `lwt` with 4.04.0+trunk, and it failed because `ppx_tools` is not yet compatible with that version. However, setting `ppx_tools` as an optional dep (because I use a copy of the opam repository that I could easily modify...), I could finally install it.

My suggestion:
- set `ppx_tools` as an optional dependency of `lwt` in opam
- create a package `lwt+ppx`, with no sources, but that depends on both `lwt` and `ppx_tools`. Packages that depend on the `ppx` syntax of `lwt` would then use a dependency on that new package instead of the one on `lwt`.

**Comment by aantron** [comment]
Not a bad suggestion, but following it exactly will have to wait, as it will be a breaking change for some users. See also https://github.com/ocsigen/lwt/issues/41.

**Comment by Drup** [comment]
I don't think that's a good idea, users expect to have the ppx on the relevant version of ocaml+lwt.

@lefessan The right solution for you is to make base-no-ppx available on 4.04, via a pin. It's a simple one liner, and it doesn't bother non-bleeding edge users. See https://github.com/ocaml/opam-repository/pull/6373

**Comment by aantron** [comment]
@Drup's suggestion is the right way.

I'm not sure how to pin `base-no-ppx` with a one-liner, but it can be done like this:

```
mkdir base-no-ppx
echo 'opam-version: "1.2"' > base-no-ppx/opam
opam pin add -y base-no-ppx base-no-ppx
```

Remove the version constraint, and `base-no-ppx` should install.
